### PR TITLE
add docs for AllowPrivilegeEscalation

### LINKED ISSUE #5239: Pod Security Policy doc is missing AllowPrivilegeEscalation parameter
**This is a...** 

- [x] Feature Request
- [ ] Bug Report

**Problem:**
The PSP doc doesn't mention [`AllowPrivilegeEscalation`/`DefaultAllowPrivilegeEscalation` fields](https://github.com/kubernetes/kubernetes/pull/47019) that were added recently.

**Proposed Solution:**
The PSP doc should list them and explain their meaning and usage.

**Page to Update:**
https://kubernetes.io/docs/concepts/policy/pod-security-policy/

CC @jessfraz @pweil-

### PR DESCRIPTION
**Pull request by jessfraz**
See https://github.com/kubernetes/features/issues/381 https://github.com/kubernetes/kubernetes/pull/47019 and https://github.com/kubernetes/community/pull/639

Closes #5239

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5448)

### CODE REVIEW

**Comment by steveperry-53** [comment]
@jessfraz, Sorry, I thought I was just setting the base to release-1.8. But now we see a whole bunch of commits and files. I don't understand what happened.

**Comment by jessfraz** [comment]
Whoops I can rebase so sorry was my bad

File: docs/concepts/policy/pod-security-policy.md [hunk 1]
```diff
@@ -124,7 +126,20 @@ configMap, downwardAPI, emptyDir, persistentVolumeClaim, secret, and projected.
...
+### AllowPrivilegeEscalation
+
+Gates whether or not a user is allowed to set the security context of a container
+to `allowPrivilegeEscalation=true`. This field defaults to `false`.
+
+### DefaultAllowPrivilegeEscalation
+
+Sets the default for the security context `AllowPrivilegeEscalation` of a container.
+This bool directly controls if the `no_new_privs` flag will be set on the
+container process. It defaults to `nil`. Setting it to `false`
```
Comment (php-coder) [inline docs/concepts/policy/pod-security-policy.md]: > It defaults to `nil`.

Could you also explain what does it mean to a user, which behavior will be?

**Comment by steveperry-53** [comment]
@jessfraz, Can you  rebase your branch again, so that it picks up 99fbc2b. That will fix the deploy/netlify failure. Thanks.

File: docs/tasks/configure-pod-container/security-context.md [hunk 2]
```diff
@@ -345,6 +347,8 @@ label given to all Containers in the Pod as well as the Volumes.
 * [Security Contexts design document](https://git.k8s.io/community/contributors/design-proposals/security_context.md)
 * [Ownership Management design document](https://git.k8s.io/community/contributors/design-proposals/volume-ownership-management.md)
 * [Pod Security Policies](/docs/concepts/policy/pod-security-policy/)
+* [AllowPrivilegeEscalation design
+  document](https://git.k8s.io/community/contributors/design-proposals/no-new-privs.md)
```
Comment (php-coder) [inline docs/tasks/configure-pod-container/security-context.md]: The link is broken now, here is a working one: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/auth/no-new-privs.md

File: docs/tasks/configure-pod-container/security-context.md [hunk 3]
```diff
@@ -24,6 +24,8 @@ a Pod or Container. Security context settings include:
 
 * [Seccomp](https://en.wikipedia.org/wiki/Seccomp): Limit a process's access to open file descriptors.
 
+* AllowPrivilegeEscalation: Controls whether a process can gain more privileges than it's parent process. This bool directly controls if the [`no_new_privs`](https://www.kernel.org/doc/Documentation/prctl/no_new_privs.txt) flag will be set on the container process. AllowPrivilegeEscalation is true always when the container is: 1) run as Privileged 2) has `CAP_SYS_ADMIN`.
```
Comment (php-coder) [inline docs/tasks/configure-pod-container/security-context.md]: > AllowPrivilegeEscalation is true always when the container is: 1) run as Privileged 2) has `CAP_SYS_ADMIN`.

I'd prefer to explicitly specify the behavior: OR or AND.

File: docs/concepts/policy/pod-security-policy.md [hunk 4]
```diff
@@ -21,22 +21,24 @@ actions that a pod can perform and what it has the ability to access. The
...
+| The use of host's PID namespace                                        | `hostPID`                                   |
+| The use of host's IPC namespace                                        | `hostIPC`                                   |
+| The SELinux context of the container                                   | [`seLinux`](#selinux)                       |
+| The user ID                                                            | [`runAsUser`](#runasuser)                   |
+| Configuring allowable supplemental groups                              | [`supplementalGroups`](#supplementalgroups) |
+| Allocating an FSGroup that owns the pod's volumes                      | [`fsGroup`](#fsgroup)                       |
+| Requiring the use of a read only root file system                      | `readOnlyRootFilesystem`                    |
+| Running of containers that allow privilege escalation from it's parent | [`allowPrivilegeEscalation`](#allowPrivilegeEscalation) |
+| Control if a process can gain more privileges than it's parent process | [`defaultAllowPrivilegeEscalation`](#defaultAllowPrivilegeEscalation) |
```
Comment (steveperry-53) [inline docs/concepts/policy/pod-security-policy.md]: No apostrophe in "its".

Running of containers that allow privilege escalation from their parents.
OR
Running of a container that allows privilege escalation from its parent.

Control whether a process can gain more privileges than its parent process.

File: docs/concepts/policy/pod-security-policy.md [hunk 5]
```diff
@@ -124,7 +126,21 @@ configMap, downwardAPI, emptyDir, persistentVolumeClaim, secret, and projected.
...
+
+### DefaultAllowPrivilegeEscalation
+
+Sets the default for the security context `AllowPrivilegeEscalation` of a container.
+This bool directly controls if the `no_new_privs` flag will be set on the
+container process. It defaults to `nil`. The default behavior of `nil` will
+allow privilege escalation so as to not break setuid binaries. Setting it to `false`
+will ensure that no child process of a container can gain more privileges than
+it's parent.
+
```
Comment (steveperry-53) [inline docs/concepts/policy/pod-security-policy.md]: controls if -> controls whether

will be set -> gets set

will allow privilege escalation -> allows privilege escalation

will ensure -> ensures

it's -> its

File: docs/tasks/configure-pod-container/security-context.md [hunk 6]
```diff
@@ -24,6 +24,8 @@ a Pod or Container. Security context settings include:
 
 * [Seccomp](https://en.wikipedia.org/wiki/Seccomp): Limit a process's access to open file descriptors.
 
+* AllowPrivilegeEscalation: Controls whether a process can gain more privileges than it's parent process. This bool directly controls if the [`no_new_privs`](https://www.kernel.org/doc/Documentation/prctl/no_new_privs.txt) flag will be set on the container process. AllowPrivilegeEscalation is true always when the container is: 1) run as Privileged OR 2) has `CAP_SYS_ADMIN`.
+
```
Comment (steveperry-53) [inline docs/tasks/configure-pod-container/security-context.md]: it's -> its

controls if -> controls whether

will be set -> gets set

**Comment by steveperry-53** [comment]
@jessfraz A few comments in the text.

File: docs/tasks/configure-pod-container/security-context.md (hunk 6 above)
Comment (jessfraz) [reply docs/tasks/configure-pod-container/security-context.md]: thanks updated!

**Pull request merged**